### PR TITLE
Add styling to links in description markdown

### PR DESCRIPTION
**Pull request by breilly2**
Fixes: #935 

Also add sanitizer

**Update 12/10**: Back to marked's built-in sanitizer (for now).

### CODE REVIEW

**Comment by zarsky-broad** [comment]
We're already sanitizing. See [here](https://github.com/DataBiosphere/saturn-ui/blob/dev/src/index.js#L10)

**Comment by breilly2** [comment]
Ah, I didn't see that. Good to know we're not neglecting it. However, escaping isn't quite enough for markdown. You can still do bad things in links, e.g.:
```
[click me](javascript:alert('BOOM'))
```
[This issue](https://github.com/markedjs/marked/issues/1232) has discussion about marked's stance on sanitization. In particular, [this comment](https://github.com/markedjs/marked/issues/1232#issuecomment-414308183) mentions some possible sanitizers to use. I just picked the first one because it had been mentioned earlier in the comments but I don't have strong convictions about that choice.

**Comment by zarsky-broad** [comment]
Indeed, that was the thread I was working from as well. I think I would have wound up with the same had the current method not done the trick. At any rate, if this swap is necessary, you should replace it in index.js, as that sets it for all uses of marked (i.e. in the preview of the editor)

File: src/pages/workspaces/workspace/Dashboard.js [hunk 1]
```diff
@@ -141,8 +141,12 @@ export const WorkspaceDashboard = _.flow(
                 options: {
                   autofocus: true,
                   placeholder: 'Enter a description',
+                  renderingConfig: {
+                    singleLineBreaks: false
```
Comment (breilly2) [inline src/pages/workspaces/workspace/Dashboard.js:145]: This avoids a bug where the preview would change the marked options used when rendering. `false` is more consistent with what FireCloud does.

**Comment by zarsky-broad** [comment]
So, is escaping sufficient in the end?

**Comment by breilly2** [comment]
Not quite. It turns out that marked _does_ currently still sanitize. However, it is implemented directly in the link renderer. My previous implementation of link styling overrode link rendering in order to supply inline styles, effectively removing link sanitization. That's why I thought it wasn't sanitizing.

For things that aren't markdown links, escaping is still necessary (and default, as it turns out; we technically don't need to specify `sanitizer: _.escape`, just `sanitize: true`). The documentation around all of this is not sufficient; I had to read the source and try a bunch of things to figure it all out.

Still to do: look into adding a git hook to remind us to be careful when upgrading marked in the future because they seem to be moving toward removing built-in sanitization (and, according to Brett, comments aren't valid in `package.json` 🙁).

**Comment by zarsky-broad** [comment]
Perhaps a good step in the meantime would be to change the version range in the package.json for marked from `^` to `~`.

File: src/components/common.js [hunk 2]
```diff
@@ -321,6 +321,15 @@ export const methodLink = (config, firecloudRoot, dockstoreRoot) => {
   return sourceRepo === 'agora' ? `${firecloudRoot}/#methods/${methodNamespace}/${methodName}/${methodVersion}` : `${dockstoreRoot}/workflows/${methodPath}`
 }
 
+/**
+ * WARNING: Be very careful when using custom renderers because they may override marked's built-in
+ * content sanitization.
+ * @param children markdown content; must be a string
```
Comment (zarsky-broad) [inline src/components/common.js]: See [the jsdoc docs](http://usejsdoc.org/tags-param.html) for how to specify the type of a param.

**Review by zarsky-broad — APPROVED** [review]
One comment.

**Comment by breilly2** [comment]
@zarsky-broad From what I can tell (based on https://nodesource.com/blog/semver-tilde-and-caret/), `^` and `~` behave the same when major == 0 and minor > 0. That page is a little old but I didn't find anything to suggest it's not still accurate. Is my interpretation wrong or is there some non-functional intent implied by the choice in cases like this?

**Comment by zarsky-broad** [comment]
It's more about communicating intent.

**Comment by breilly2** [comment]
Thanks. Bonus, `~` doesn't have any special major-0 rules.

**Pull request merged**